### DISCUSSION TITLE
需要从UMI 转化为CRA,有没有文档可遵循?

### DISCUSSION THREAD

**Discussion by gaohongwei** [umijs/umi#9818] [General]
主要原因是UMI的 APP，定制化受限。例如，增加Provider，动态菜单。

**Comment by fz6m** [comment]
有哪些问题，描述一下，提供一个复现看看。

**Reply by gaohongwei** [reply]
[fz6m](https://github.com/fz6m)， 例如，
根据用户权限，确立、显示左边的菜单
根据当前的用户选择，确立、显示左边的菜单。 例如，选择了项目，才有项目的子菜单，没选择项目，就没有项目的子菜单。
还有就是，有时需要增加应用级别的Provider。

**Reply by fz6m** [reply]
这些你用 umi max 就都有了，可以创建一个新项目看看。

 - 控制菜单的权限 ：https://umijs.org/docs/max/access

 - 在 `src/app.tsx` 添加 provider ：https://umijs.org/docs/api/runtime-config#rootcontainerlastrootcontainer-args

**Reply by gaohongwei** [reply]
谢谢 [fz6m](https://github.com/fz6m) 。 umijs 定义菜单，路由，短路径都很方便。所以，早起是使用umijs 。 但是总觉得定制化化门槛有点高。有点想换掉。试了两个星期，没换掉。
